doc/user_manual/rom.tex

```diff
@@ -191,3 +191,71 @@ \subsection{ROM}
+The capabilities implemented enable to retrieve sampled and trained ROMs, pickle (serialize), and output python files as
+defined in the \xmlNode{IOStep}.
+
+The trained file is copied and processed separately to output two files required to run
+Pyomo. The python output file contains the infrastructure inheriting the External Grey Box Model and others included in
+the Pyomo contributions package Pynumero to print a file with Pyomo syntax. The `.pk' output file is the pickled version of the
+trained ROM and can be ran with the python output file.
+
+Since the code developed in ROMs is not a subtype, the code activation is triggered by file type. To ensure developed Pyomo code
+generator is utilized in ROMs, one of the file extensions in the output of the \xmlNode{IOStep} must be `.py'. The subtype of the
+ROM model can be of any kind and is the user's decision to choose the training method.
+
+When loading ROMs from file, the user is expected to know the I/O of a ROM and loading of the ROM file must use \xmlNode{IOStep}.
+
+To verify the output files in the given example below are runnable with GrayBoxModel, several optional RAVEN dependencies are required.
+The following are the required optional libraries: pyomo 6.4 or over, cmake, glpk, ipopt, cyipopt, and pyomo-extensions. The syntax to run
+trained ROMs and GayBoxModel is the following:
+
+\begin{lstlisting}
+  python3 python_output_file.py -r rom_pickle_file.pk -f location_of_raven_framework
+\end{lstlisting}
+
+If the user does not have a setup pyomo and pynumero environment, it is recommended to activate RAVEN libraries using the following command
+on the terminal:
+
+\begin{lstlisting}
+  source scripts/establish_conda_env.sh --load
+\end{lstlisting}
+
+\textbf{Example:} For this example the external model is trained and further loaded as `out\_rom'. The loaded trained model is separately
+processed to `rom\_out.py' and `rom\_pickle.pk'.
+\begin{lstlisting}[style=XML,morekeywords={name,subType}]
+  <Simulation>
+    ...
+    <Files>
+      <Input name="rom_out.py" type="">rom_out.py</Input>
+      <Input name="rom_pickle.pk" type="">rom_pickle.pk</Input>
+    </Files>
+    ...
+    <Steps>
+      ...
+      <MultiRun name="sample">
+        ...
+        <Model class="Models" type="ExternalModel">attenuate</Model>
+        ...
+        <Output class="DataObjects" type="PointSet">samples</Output>
+      </MultiRun>
+      <RomTrainer name="train">
+        <Input class="DataObjects" type="PointSet">samples</Input>
+        <Output class="Models" type="ROM">out_rom</Output>
+      </RomTrainer>
+      <IOStep name="serialize">
+        <Input class="Models" type="ROM">out_rom</Input>
+        <Output class="Files" type="">out_rom.py</Output>
+      </IOStep>
+      <IOStep name="pickle">
+        <Input class="Models" type="ROM">out_rom</Input>
+        <Output class="Files" type="">out_pickle.pk</Output>
+      </IOStep>
+      ...
+    </Steps>
+    ...
+  </Simulation>
+\end{lstlisting}
```

Comment:
An INL fellow has been working on the documentation and I have been told it would be ready next week since he has pending work from another project. It is in word format and will need to be converted into Latex syntax.

Reply:
Done.